crates/ide_db/src/lib.rs

```diff
@@ -148,6 +148,7 @@ pub enum SymbolKind {
     Local,
     Macro,
     Module,
+    Region,
```

Comment:
Hm, I think adding `Region` to SymbolKind would be wrong, as it is unlike other symbols. Let's rather introduce 

```
enum StructureNodeKind { SymbolKind(SymbolKind), Region }
```

Reply:
I'm not sure here. If we separate region from others because it's a token, then we could provide a separate enum for symbols that are based on tokens (there might be more than just one in future)

Reply:
Hm, not sure if it makes sense to treat Region as something even remotely resembling a symbol: it exists completely outside of the language definition. So I'd rather keep this as a special case for structure. Generalizing later shouldn'b be hard. In contrast, un-generalizing would be harder, as every user of symbol would care about region.

Reply:
Implemented it. I didn't replace all usages of `SymbolKind` for `StructureNodeKind::SymbolKind`, but instead changed only the ones, that somehow call `to_proto::symbol_kind`